### ISSUE TITLE
Error with simple photomaker workflow

### ISSUE THREAD

**Issue by airtisticchad** [comfyanonymous/ComfyUI#2639]
Error occurred when executing PhotoMakerEncode:

Given normalized_shape=[4096], expected input with shape [*, 4096], but got input of size[1, 2816]

File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\execution.py", line 155, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\execution.py", line 85, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\execution.py", line 78, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_photomaker.py", line 175, in apply_photomaker
out = photomaker(id_pixel_values=pixel_values.unsqueeze(0), prompt_embeds=cond.to(photomaker.load_device),
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_photomaker.py", line 113, in forward
updated_prompt_embeds = self.fuse_module(prompt_embeds, id_embeds, class_tokens_mask)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_photomaker.py", line 85, in forward
stacked_id_embeds = self.fuse_fn(image_token_embeds, valid_id_embeds)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_photomaker.py", line 52, in fuse_fn
stacked_id_embeds = self.mlp1(stacked_id_embeds) + prompt_embeds
^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_photomaker.py", line 34, in forward
x = self.layernorm(x)
^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
return self._call_impl(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
return forward_call(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy\ops.py", line 86, in forward
return self.forward_comfy_cast_weights(*args, **kwargs)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\ComfyUI\comfy\ops.py", line 82, in forward_comfy_cast_weights
return torch.nn.functional.layer_norm(input, self.normalized_shape, weight, bias, self.eps)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\ai\comfy\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\functional.py", line 2543, in layer_norm
return torch.layer_norm(input, normalized_shape, weight, bias, eps, torch.backends.cudnn.enabled)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
![image](https://github.com/comfyanonymous/ComfyUI/assets/139369799/eebf4b77-485f-47e8-8d12-67fa1aa4cf1a)

**Comment by Arron17** [comment]
That looks like an SD 1.5 model. Photomaker only works with SDXL.

**Comment by airtisticchad** [comment]
> That looks like an SD 1.5 model. Photomaker only works with SDXL.

Oh okay my bad.

**Comment by ltdrdata** [comment]
@Joao393112 

![Image](https://github.com/user-attachments/assets/4f51b543-b16d-41d9-896e-276b7551623d)

Invalid clip vision model is used.
You have to use `clip_vision_h.safetensors`

![Image](https://github.com/user-attachments/assets/df53f2c8-59f8-40d9-bb16-dcc5ad964255)
